fedimint-api/src/db/mod.rs

```diff
@@ -400,12 +496,17 @@ mod tests {
     }
 
     #[derive(Debug, Encodable, Decodable)]
-    struct TestKey(u64);
+    pub struct TestKeyV1 {
+        pub value: u64,
+    }
 
-    impl DatabaseKeyPrefixConst for TestKey {
-        const DB_PREFIX: u8 = TestDbKeyPrefix::Test as u8;
-        type Key = Self;
-        type Value = TestVal;
+    impl TestKeyV1 {
+        pub fn migrate(&self) -> TestKeyV2 {
```

Comment:
I haven't been able to figure out how to make this into a trait so that any DatabaseKey or DatabaseValue can implement it. The workaround I came up with for now is to just to have a macro that calls migrate, so if you want to use the macro you'll need to implement migrate.